mirobo/vacuum_cli.py

```diff
@@ -294,27 +294,59 @@ def fanspeed(vac: mirobo.Vacuum, speed):
         click.echo("Current fan speed: %s" % vac.fan_speed())
 
 
-@cli.command()
-@click.argument('timer', required=False, default=None)
+@cli.group(invoke_without_command=True)
 @pass_dev
-def timer(vac: mirobo.Vacuum, timer):
-    """Schedule vacuuming, times in GMT."""
-    if timer:
-        raise NotImplementedError()
-        # vac.set_timer(x)
-    else:
-        timers = vac.timer()
-        for idx, timer in enumerate(timers):
-            color = "green" if timer.enabled else "yellow"
-            #  Note ts == ID for changes
-            click.echo(click.style("Timer #%s, id %s (ts: %s)" % (
-                idx, timer.id, timer.ts), bold=True, fg=color))
-            print("  %s" % timer.cron)
-            min, hr, x, y, days = timer.cron.split(' ')
-            # hr is in gmt+8 (chinese time), TODO convert to local
+@click.pass_context
+def timer(ctx, vac: mirobo.Vacuum):
+    """List and modify existing timers."""
+    if ctx.invoked_subcommand is not None:
+        return
+    timers = vac.timer()
+    for idx, timer in enumerate(timers):
+        color = "green" if timer.enabled else "yellow"
+        #  Note ts == ID for changes
+        click.echo(click.style("Timer #%s, id %s (ts: %s)" % (
+            idx, timer.id, timer.ts), bold=True, fg=color))
+        print("  %s" % timer.cron)
+        min, hr, x, y, days = timer.cron.split(' ')
+        # hr is in gmt+8 (chinese time), TODO convert to local
+        if hr != '*':
             hr = (int(hr) - 8) % 24
-            cron = "%s %s %s %s %s" % (min, hr, x, y, days)
-            click.echo("  %s" % pretty_cron.prettify_cron(cron))
+        cron = "%s %s %s %s %s" % (min, hr, x, y, days)
+        click.echo("  %s" % pretty_cron.prettify_cron(cron))
+
+
+@timer.command()
+@click.option('--cron')
+@click.option('--command', default='', required=False)
+@click.option('--params', default='', required=False)
+@pass_dev
+def add(vac: mirobo.Vacuum, cron, command, params):
+    """Schedule vacuuming.."""
+    click.echo(vac.add_timer(cron, command, params))
+
+
+@timer.command()
+@click.argument('timer_id', type=int, required=True)
+@pass_dev
+def delete(vac: mirobo.Vacuum, timer_id):
+    """Delete existing schedule."""
+    click.echo(vac.delete_timer(timer_id))
+
+
+@timer.command()
+@click.argument('timer_id', type=int, required=True)
+@click.option('--on', is_flag=True)
+@click.option('--off', is_flag=True)
+@pass_dev
+def update(vac: mirobo.Vacuum, timer_id, on, off):
+    """Update (on or off) an existing scheduled."""
```

Comment:
Enable/Disable a stored timer. I would prefer "--enable" and "--disable" as well.

Reply:
Changed to `--enable` and `--disable`, """Enable/disable a timer.""" as help message to be consistent with other commands.